bin/init.sh

```diff
@@ -123,9 +123,6 @@ if [ ! -f "$ISTIO_ENVOY_DEBUG_PATH" ] || [ ! -f "$ISTIO_ENVOY_RELEASE_PATH" ] ;
     # Download debug envoy binary.
     mkdir -p $ISTIO_ENVOY_DEBUG_DIR
     pushd $ISTIO_ENVOY_DEBUG_DIR
-    if [ "$LOCAL_OS" == "darwin" ]; then
```

Comment:
These were apparently added by https://github.com/istio/istio/commit/60c35a4cafef207d580b45006f9cf36614416532

Is removing these going to break that?

Reply:
Yes that could be.
Maybe we should add a env var to indicate the difference between local test and docker build?

Reply:
@kimikowang Please do not remove these checks, since we cannot build envoy for MAC from the istio repo, we need to download it from a specific location.

Reply:
I suppose we want to make the followings work:

1. Building and testing binaries on mac. It looks like that's what https://github.com/istio/istio/commit/60c35a4cafef207d580b45006f9cf36614416532 was about. I suppose this does not involve "make docker push".

2. The same as #1 on linux.

3. Building images on any platform. I suppose we need the linux version anyway....

So instead of introducing yet another env var, how about making it command/context sensitive?

Reply:
Please review the next line. I added one more condition for this check.

Reply:
I set the env var in bin/testEnvLocalK8S.sh and modify the check condition. I think this is command and context sensitive? @hklai

Reply:
How about the other way around?

In the docker/push command, set a variable that can be used to override the OS check.

 if [ "$LOCAL_OS" == "darwin" ] && [ "$USE_DOCKER" != "true" ]; then


It does not require the users (e.g. testEnvLocalK8S.sh and test_setup.sh) to set up anything and seems to be more future proof. Can that be done?